libs/payments/ui/src/lib/client/components/Header/index.tsx

```diff
@@ -0,0 +1,374 @@
+                    </LinkExternal>
+                  </Localized>
+                </div>
+              </div>
+            </div>
+          )}
+        </div>
+        {/** Avatar Menu, TODO: convert to Radix Primitive as part of FXA-11035 */}
+        {signedIn && (
+          <div className="relative" ref={avatarMenuInsideRef}>
+            <Localized id="avatar-menu">
+              <button
+                onClick={toggleAvatarMenuRevealed}
+                title="Mozilla account menu"
+              >
+                <Localized id="avatar-menu-icon">
+                  <Image
+                    unoptimized={true}
+                    src={auth.user?.image ?? defaultAvatarIcon}
+                    alt="Account profile picture"
+                    width="10"
+                    height="10"
+                    className="w-10 rounded-full"
+                  />
+                </Localized>
+              </button>
+            </Localized>
+
+            {/** Avatar Dropdown */}
+            {isAvatarMenuRevealed && (
+              <div
+                dir="ltr"
+                className="w-64 bg-white absolute shadow-md rounded-lg ltr:-left-52"
+                role="menu"
+              >
+                <div className="flex flex-wrap">
+                  <div className="flex w-full p-4 items-center">
+                    <div className="ltr:mr-3 flex-none">
+                      <Localized id="avatar-menu-icon">
+                        <Image
+                          unoptimized={true}
+                          src={auth.user?.image ?? defaultAvatarIcon}
+                          alt="Account profile picture"
+                          width="10"
+                          height="10"
+                          className="w-10 rounded-full"
+                        />
+                      </Localized>
+                    </div>
+                    <p className="leading-5 max-w-full truncate">
+                      <Localized id="drop-down-menu-signed-in-as">
+                        <span className="text-grey-400 text-xs">
+                          Signed in as
+                        </span>
+                      </Localized>
+                      <span className="font-bold block truncate">
+                        {/* eslint-disable @typescript-eslint/no-non-null-assertion */}
+                        {auth.user!.name || auth.user!.email}
+                      </span>
+                    </p>
+                  </div>
+                  <div className="w-full">
+                    <div className="bg-gradient-to-r from-blue-500 via-pink-700 to-yellow-500 h-px" />
+                    <div className="px-4 py-5">
+                      <button
+                        onClick={async () => {
+                          const redirectUrl = await auth.signOut(routeParams);
+                          router.push(redirectUrl);
+                          window.location.href = redirectUrl;
```

Comment:
This bit of hacky magic is required to work around the fact that a redirect to /new from the auth signout hook causes the browser to attempt to render /new as if it were a page (and not a redirect route). Both `router.push` and `window.location.href = redirectUrl` are required here. Without `router.push` we see an unacceptable flash as the browser attempts (and fails) to render /new. Without `window.location.href = redirectUrl`, the page itself refreshes, but the header doesn't flush its `session` object from the `signOut` call, and so the session persists.

Reply:
@StaberindeZA my theory is that there is a race condition somewhere that is necessitating this (and preventing simply redirecting to /new from working), but I haven't for the life of me been able to track it down. Let me know if you have any thoughts, or if this pattern rings any bells for you!